### PR TITLE
fix(mcdu): revise route reserve calculation

### LINKED ISSUE #4093: Route reserved fuel not included in block/extra fuel calculation
**Mod Version**

```
{
  "built": "2021-03-23T03:51:43+00:00",
  "ref": "master",
  "sha": "9dfce11f233cfdbf862024bcd5e22f362a8d2c03",
  "actor": "Benjozork",
  "event_name": "manual"
}
```

**Describe the bug**

I think have a spotted an inconsistency in the calculated block fuel value on the fuel prediction page in what I'm seeing in the simulator vs various other references.

In the sim, the default block fuel is calculated as: `BLOCK = TAXI + TRIP + MIN DEST FOB` (where `MIN DEST = ALTN + FINAL` by default). When entering a block fuel value manually, any value that is in excess of the default calculated value is added as `EXTRA`. In this case, extra fuel will be calculated as: `EXTRA = BLOCK - TAXI - TRIP - MIN DEST FOB`).

*Screenshot showing the automatically calculated block fuel value*
insim_automatic_block

*Screenshot with a manually entered block fuel value*
insim_manual_block

However, when comparing this to other sources I think I'm seeing something different in that the route reserved fuel is also factored into this calculation.

In both the Honeywell Pilot's Guide and the cockpit recording linked below, it seems that `BLOCK = TAXI + TRIP + RTE RSV + MIN DEST FOB + EXTRA `, i.e. the route reserved fuel is included in the automatic block fuel calculation (or the extra fuel calculation if the block fuel is manually entered).

The referenced paragraph from the FCOM also seems to confirm that the route reserved fuel should be included in the calculation such that extra is calculated as: `EXTRA = BLOCK - TAXI - TRIP - RSV - MIN DEST FOB`

I am aware that the route reserved fuel is 'converted' to extra fuel after takeoff, however my understanding is that this is just a difference in how it is displayed. Shouldn't the numbers during the preflight initialization all add up like they seemingly do in the references below?

Of course it's also possible (or perhaps likely) that I'm just misunderstanding something - if so I would be happy to be corrected :)

**To Reproduce**

1. Enter FROM/TO and ALTN on INIT A. The settings I used to test this are seen in the screenshot below.
2. Check both the calculated block fuel value using the fuel planner and the calculated extra fuel value when using a manually entered block fuel value  on INIT B

insim_init_a

**Expected behavior**

I am not 100% sure, but my understanding is that the automatically calculated block fuel value should be: `BLOCK = TAXI + TRIP + RSV + MIN DEST FOB`
And when entering a manual block fuel value, extra should be calculated as: `EXTRA = BLOCK - TAXI - TRIP - RSV - MIN DEST FOB`

**Actual behavior**

The automatically calculated block fuel is: `BLOCK = TAXI + TRIP + MIN DEST FOB`
When entering a manual block fuel value, extra is calculated as: `EXTRA = BLOCK - TAXI - TRIP - MIN DEST FOB`

**References**

These sources seem to include the route reserved fuel in the block/extra calculation:

*Screenshot showing the INIT B page from this Youtube video:* https://www.youtube.com/watch?v=DP8soiH_4jE&t=122s
cockpit_view

*Honeywell Pilot's Guide*
pilots_guide

*CCM Airlines FCOM*
fcom

**Additional context**

Was this working before/when did the issue start occurring?

Unknown

Is this a problem in the vanilla unmodded game?

Unknown

Discord username (if different from GitHub): Jethro#4431

### PR DESCRIPTION
**Pull request by Lucky38i**
Fixes #4093

## Summary of Changes

- Route reserve weight is now calculated from trip weight rather than block fuel.
- retrieving extra fuel now correctly removes or adds route reserve weight to it's calculation depending on if the plane is flying or not.
- Minimum Destination FOB now update at the correct sequence when using fuel planning

## References

- Pegasus Step 1A (Rev 0), 2009 edition manual

## Additional context

Discord username (if different from GitHub):
Lucky38i
## Testing instructions

For Testers, verify the following:
- After entering block fuel/using fuel planning, try to change the reserve fuel percentage, confirm the extra fuel changes
- MinDestFob is Alternate Fuel + Final Fuel.
- After entering block fuel, default route reserve weight is 5% of trip weight, if not ignore, this is instead 5 minutes of holding fuel.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -230,6 +230,7 @@
 1. [EFB] Set font of OFP to monospace - @chkgk (Chriskgk#4359)
 1. [SOUND] Added electrical system sounds - @hotshotp (Boris)
 1. [SOUND] General improvements to cockpit and cabin ambience - @hotshotp (Boris)
+1. [MCDU] Revised Reserve Weight Calculation and Default weight rule-set added (Lucky38i)
```
Comment (pepperoni505) [inline .github/CHANGELOG.md:233]:
```suggestion
1. [MCDU] Revised Reserve Weight Calculation and Default weight rule-set added - @Lucky38i (Lucky38i)
```

**Comment by Samueleonard** [comment]
Category: QA Trial
Name: 1samueleonard1#0001
Date of testing: 24/03/2021
Version of the sim: 1.14.5.0
PR Tested: #4103
Tier of testing: 1
Changes to observe: rsv fuel %, mindestfob, general init B functionality
Testing technique: Tryout

Steps you've made to test the Pull Request:
1. spawned in egkk - eddb
2. entered fpln and autofilled init b   (6.4t fuel - 59,0zfw)
3. changed rte rsv % - value changed as expected
4. checked min dest fuel calc - was correct
5. checked rte rsv weight - was 0.1

Issues (If Any):

im not sure if theres a limit, or maybe a bug, but when entering some values into rte rsv/% (eg /20 for 20%) nothing happens, maybe there should be a scratchpad message?
Steps to achieve the issue

Overall Rating: Potentially To Be Improved but good

Conclusions of the testing: 
i cannot vouch for accuracy, but functionally it seems okay

**Comment by Lucky38i** [comment]
> Category: QA Trial
> Name: 1samueleonard1#1
> Date of testing: 24/03/2021
> Version of the sim: 1.14.5.0
> PR Tested: #4103
> im not sure if theres a limit, or maybe a bug, but when entering some values into rte rsv/% (eg /20 for 20%) nothing happens, maybe there should be a scratchpad message?

Hi Samuel thanks for the QA. If you enter a route reserve percentage that is out of range (bigger than 15% or small than 0%) then you will just get dashed results, no scratchpad message.
Screenshot 2021-03-24 at 16 11 53

**Review by MisterChocker — APPROVED** [review]
LGTM

**Comment by TheSquirtingPantin** [comment]
Hello All, 

Category: QA Trial
Name: Discord User Rémi#8871
Date of testing: Current date 25/03/2021
Version of the sim: V 1.14.5.0
PR Tested: #4103
Tier of testing: 1
Changes to observe:
Logic of INIT FUEL PRED RTE RSV

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. C/D at KDEN
2. Check for void and logic for RTE RSV entry on INIT FUEL PRED
3. 
5.

Flight notes (When done a full flight):N/A

Media: N/A

Issues (If Any):
	- Entry 3L cannot be done in weight (which I believe is normal and due to MSFS limitation but need confirmation. Testing was done only on percent value as per PR request)
	- For a given trip fuel of 4.7, the rounded value for RTE RSV (if we select 10% RTE RSV) seems to be done incorrectly. Shouldn't it be 0.4 rather than 0.5  for safety purpose? (Trying to find some reference as well on the round policy for this value)

Overall Rating: Good
Conclusions of the testing: Behavior is as expected. Minor logic adjustment on rounded value to be confirmed/Corrected. Good Job!

**Comment by Lucky38i** [comment]
> Hello All,
> 
> Category: QA Trial
> Name: Discord User Rémi#8871
> Date of testing: Current date 25/03/2021
> Version of the sim: V 1.14.5.0
> PR Tested: #4103
> - For a given trip fuel of 4.7, the rounded value for RTE RSV (if we select 10% RTE RSV) seems to be done incorrectly. Shouldn't it be 0.4 rather than 0.5 for safety purpose? (Trying to find some reference as well on the round policy for this value)
> 
> Overall Rating: Good
> Conclusions of the testing: Behavior is as expected. Minor logic adjustment on rounded value to be confirmed/Corrected. Good Job!

Thanks the uh.. @TheSquirtingPantin. There's no references in the manual of wether it's rounded conservatively or normally. I've also spoken to an a320 pilot and he's not sure either. For the time being it'll remain on the normal side until a reference comes around.

**Review by beheh — APPROVED** [review]

**Pull request merged**